infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java

```diff
@@ -55,230 +54,283 @@
+            throw new IllegalArgumentException("AWS secret access key must be specified");
         }
-
-        if (parameters[parameterIndex++] == null) {
-            throw new IllegalArgumentException("The image id must be specified");
+        parameterIndex++;
+        // numberOfInstances
+        if (parameters[parameterIndex] == null) {
+            throw new IllegalArgumentException("The number of instances to create must be specified");
         }
-
-        // VM username
+        parameterIndex++;
+        // numberOfNodesPerInstance
         if (parameters[parameterIndex] == null) {
-            parameters[parameterIndex++] = "";
+            throw new IllegalArgumentException("The number of nodes per instance to deploy must be specified");
         }
-
-        // key pair name
+        parameterIndex++;
+        // image
         if (parameters[parameterIndex] == null) {
-            parameters[parameterIndex++] = "";
+            parameters[parameterIndex] = DEFAULT_IMAGE;
         }
-
-        // private key file
+        if (!parameters[parameterIndex].toString().contains("/")) {
+            throw new IllegalArgumentException(String.format("Invalid image %s (image should be in format 'region/ami-id').",
+                                                             parameters[parameterIndex]));
+        }
+        parameterIndex++;
+        // vmUsername
         if (parameters[parameterIndex] == null) {
-            parameters[parameterIndex++] = "";
+            parameters[parameterIndex] = DEFAULT_VM_USERNAME;
         }
-
-        if (parameters[parameterIndex++] == null) {
-            throw new IllegalArgumentException("The number of instances to create must be specified");
+        parameterIndex++;
+        // vmKeyPairName
+        if (parameters[parameterIndex] == null) {
+            parameters[parameterIndex] = "";
         }
-
-        if (parameters[parameterIndex++] == null) {
-            throw new IllegalArgumentException("The number of nodes per instance to deploy must be specified");
+        parameterIndex++;
+        // vmPrivateKey
+        if (parameters[parameterIndex] == null) {
+            parameters[parameterIndex] = "";
         }
-
-        if (parameters[parameterIndex++] == null) {
-            throw new IllegalArgumentException("The download node.jar command must be specified");
+        parameterIndex++;
+        // ram
+        if (parameters[parameterIndex] == null) {
+            parameters[parameterIndex] = DEFAULT_RAM;
         }
-
+        parameterIndex++;
+        // cores
         if (parameters[parameterIndex] == null) {
-            parameters[parameterIndex++] = "";
+            parameters[parameterIndex] = DEFAULT_CORES;
         }
-
-        if (parameters[parameterIndex++] == null) {
-            throw new IllegalArgumentException("The amount of minimum RAM required must be specified");
+        parameterIndex++;
+        //        // spotPrice
```

Comment:
can you leave a comment why this parameter is removed and the code commented out
(maybe a todo)

Reply:
Ok, the reason was commented in the declaration. I just add "TODO" to mark the related codes.